### DISCUSSION TITLE
Issue with fetching the latest PR commits

### DISCUSSION THREAD

**Discussion by vamche** [community/community#21553] [New to GitHub]
<p>There seems to be an intermittent issue while fetching the latest commits of a PR. I tried a couple of methods, but both fail intermittently. </p>
<p>Method 1:</p>
<pre><code>git clone --depth 1 https://github.com/qawoody/welcome-pipelines.git welcome-pipelines
cd welcome-pipelines
git checkout -b master-tmp
git fetch --force --depth 1 origin "pull/4/merge:master"
git checkout master
git log --pretty=oneline
</code></pre>
<p>Method 2:</p>
<pre><code>git clone --depth 50 https://github.com/qawoody/welcome-pipelines.git welcome-pipelines
cd welcome-pipelines
git fetch origin +refs/pull/4/merge:
git checkout -qf FETCH_HEAD
git log --pretty=oneline
</code></pre>
<p>In both the cases, you will observe that the merged branch doesn’t have the latest commits. Observing this issue from the past week.</p>
<p>Can someone help me on this? Or did anyone face this problem?</p>
<p>PS: The above-mentioned repository is an example public repo. I am facing this with many other repositories. </p>
<p>Thanks</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>Thanks for asking!</p>
<p>It looks like you’re trying to get the latest commits on a PR by fetching the merge ref for the PR. Unfortunately, while the existence of the PR refs is well-known, we don’t make any guarantees about their operation. When a PR is updated, the merge ref is updated lazily and some changes may have happened in the last week that have made it update more lazily than in the past.</p>
<p>What is your ultimate goal here? Would using the <a href="https://developer.github.com/v3/pulls/#get-a-single-pull-request" rel="nofollow noopener">GitHub API</a> be a more reliable way of getting the information you’re looking for?</p>

**Comment by vamche** [comment]
<p><a class="mention" href="/lee-dohm">@lee-dohm</a> Thanks for the reply.</p>
<p>The goal is to pull the latest changes of a PR and build (CI similar to Travis CI). We have webhooks attached to the repo, which triggers builds on push/PR commits.</p>
<p>Could you please tell me if there is a recommended way to get a PR’s latest changes? (into a temporary branch)</p>

**Comment by lee-dohm** [comment]
<blockquote>
<p><a class="mention" href="/vamche">@vamche</a> wrote:</p>
</blockquote>
<blockquote>
<p>Could you please tell me if there is a recommended way to get a PR’s latest changes? (into a temporary branch)</p>
</blockquote>
<p>For that, you can see <a href="https://github.community/t/checkout-a-branch-from-a-fork/276/2">my answer on this other topic</a>. You’ll notice that it uses the head ref instead of the merge ref.</p>

**Comment by vamche** [comment]
<p><a class="mention" href="/lee-dohm">@lee-dohm</a> Thank you. Using head ref solved the issue.</p>
